### DISCUSSION TITLE
How to enable admin page on vaultwarden?

### DISCUSSION THREAD

**Discussion by amit510** [tteck/Proxmox#250] [Q&A] [answered]
I was trying to locate the .env file in order to enable the admin page. I tried few default locations, but still not able to enable the admin page. Most of the instructions out there are for docker installs. 

https://vault.domain.com/admin shows the following: 
The admin panel is disabled, please configure the 'ADMIN_TOKEN' variable to enable it

Could someone please point to the folder location where I can access this .env file? 

Thanks.

**Comment by tteck** [comment]
environment file path `/opt/vaultwarden/.env` after 2022-05-29
`/etc/vaultwarden.env` before 2022-05-29

**Reply by amit510** [reply]
@tteck It worked! thank you for the prompt reply! Appreciate your help.

**Reply by tteck** [reply]
when you register the first account, it's admin IINM

**Reply by schiz0phr3ne** [reply]
> when you register the first account, it's admin IINM

Nope, administration is reachable with only a token.
No need to be an user here.

**Comment by mikoning** [comment]
Im sorry if this is a dumb question, but where do I enter this location?

**Reply by tteck** [reply]
IP:8000/admin

**Reply by mikoning** [reply]
Ah sorry, that part I understood. I ment the path for the environment variables. Should I look for "/opt/vaultwarden/.env" in the console of the lxc?

**Reply by tteck** [reply]
yes, `cat /opt/vaultwarden/.env`

**Reply by mikoning** [reply]
Thanks for the quick respones! The file didnt exist, so I made a new one based on an example, uncommented the ADMIN_TOKEN row, but no result unfortunately.

The original problem is that Im unable to add a new organisation. When I try to that there is a loading page which wont go on. Googling this told me to go to the admin page which is disabled. Some more googling got me to this page (https://github.com/dani-garcia/vaultwarden/discussions/2802) saying something about incompatible client and web versions. Could this be the case here as well? If you want me to open a new thread for this, let me know!

**Reply by tteck** [reply]
If it's an older install (before 2022-05-29) the path is `cat /etc/vaultwarden.env`

**Reply by mikoning** [reply]
I installed after that date, but tried it anyway (copying the .env file to that directory)...no luck unfortunately

**Reply by tteck** [reply]
I don't know what has happened to your install, but the script creates
```
cat <<EOF > /opt/vaultwarden/.env
ADMIN_TOKEN=$(openssl rand -base64 48)
ROCKET_ADDRESS=0.0.0.0
DATA_FOLDER=/opt/vaultwarden/data
DATABASE_MAX_CONNS=10
WEB_VAULT_FOLDER=/opt/vaultwarden/web-vault
WEB_VAULT_ENABLED=true
EOF
```
Maybe it's time to export all of your data, reinstall Vaultwarden and import data.

**Reply by mikoning** [reply]
Sure no problem. I will let you know the outcome. 

Thanks for your help, really appreciate it!

**Reply by mikoning** [reply]
The reinstall fixed my problem. Thanks again!

**Comment by R0yk3** [comment]
Hi, 
In Alpine it works different

/etc/conf.d/vaultwarden <-- this is a configuration file.
Set it in the configuration file!
find

> Token for the admin interface, preferably use a long random string
 One option is to use 'openssl rand -base64 48'
 If not set, the admin panel is disabled
export ADMIN_TOKEN='Put here your secret key'

reboot and it works.
is this the way it should be or doing something wrong?
i get the message ou are using a plain text `ADMIN_TOKEN` which is insecure.
so iguess it is not perfect.

**Comment by tteck** [comment, marked as answer]
![Screenshot 2023-06-19 2 40 28 PM](https://github.com/tteck/Proxmox/assets/21342670/8c3411ca-f00e-4d71-b959-bd43ee26a9c7)

**Reply by R0yk3** [reply]
Thank YOU, this is nicer than RTFM.
